pymc3/tests/test_distributions_random.py

```diff
@@ -491,43 +414,364 @@ class TestMoyal(BaseTestCases.BaseTestCase):
     params = {"mu": 0.0, "sigma": 1.0}
 
 
-@pytest.mark.skip(reason="This test is covered by Aesara")
-class TestCategorical(BaseTestCases.BaseTestCase):
-    distribution = pm.Categorical
-    params = {"p": np.ones(BaseTestCases.BaseTestCase.shape)}
+class BaseTestDistribution(SeededTest):
+    pymc_dist: Optional[Callable] = None
+    pymc_dist_params = dict()
+    reference_dist: Optional[Callable] = None
+    reference_dist_params = dict()
+    expected_rv_op_params = dict()
+    tests_to_run = []
+    size = 15
+    decimal = 6
+
+    sizes_to_check: Optional[List] = None
+    sizes_expected: Optional[List] = None
+    repeated_params_shape = 5
+
+    def test_distribution(self):
+        self._instantiate_pymc_rv()
+        if self.reference_dist is not None:
+            self.reference_dist_draws = self.reference_dist()(
+                **self.reference_dist_params, size=self.size
+            )
+        for test_name in self.tests_to_run:
+            self.run_test(test_name)
+
+    def run_test(self, test_name):
+        {
+            "check_pymc_dist_matches_reference": self._check_pymc_draws_match_reference,
+            "check_pymc_params_match_rv_op": self._check_pymc_params_match_rv_op,
+            "check_rv_size": self._check_rv_size,
+        }[test_name]()
+
+    def _instantiate_pymc_rv(self, dist_params=None):
+        params = dist_params if dist_params else self.pymc_dist_params
+        with pm.Model():
+            self.pymc_rv = self.pymc_dist(
+                **params,
+                size=self.size,
+                rng=aesara.shared(self.get_random_state(reset=True)),
+                name=f"{self.pymc_dist.rv_op.name}_test",
+            )
 
-    def get_random_variable(
-        self, shape, with_vector_params=False, **kwargs
-    ):  # don't transform categories
-        return super().get_random_variable(shape, with_vector_params=False, **kwargs)
+    def _check_pymc_draws_match_reference(self):
+        # need to re-instantiate it to make sure that the order of drawings match the reference distribution one
+        self._instantiate_pymc_rv()
+        assert_array_almost_equal(
+            self.pymc_rv.eval(), self.reference_dist_draws, decimal=self.decimal
+        )
 
-    def test_probability_vector_shape(self):
-        """Check that if a 2d array of probabilities are passed to categorical correct shape is returned"""
-        p = np.ones((10, 5))
-        assert pm.Categorical.dist(p=p).random().shape == (10,)
-        assert pm.Categorical.dist(p=p).random(size=4).shape == (4, 10)
-        p = np.ones((3, 7, 5))
-        assert pm.Categorical.dist(p=p).random().shape == (3, 7)
-        assert pm.Categorical.dist(p=p).random(size=4).shape == (4, 3, 7)
+    def _check_pymc_params_match_rv_op(self) -> None:
+        try:
```

Comment:
Do we need this `try / except` block? The test will just fail if this happens, and I don't see why it would happen in the first place (?)

Reply:
true, removing it